### PR TITLE
fix bugs in front-end

### PR DESCRIPTION
**Pull request by QingyangKong**
Add history to allow user transfer in pages.
Add some verification to indicate users' invalid input.
Fix the issue that clubs are not displayed completely.
Fix the crash caused by users' invalid input.
Fix club list sorting issues.

### CODE REVIEW

File: src/stores/activityStore/index.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ class ActivityStore {
       const authorAvatar = constructPicUrl(player['icon'])
       const authorName = player['name']
       const clubName = await dataContract.methods.clubName().call()
+      const isForbidden = eventInfo['punished']
```
Comment (Keith-CY) [inline src/stores/activityStore/index.js]: why not `eventInfo.punished`

File: src/stores/activityStore/index.js [hunk 2]
```diff
@@ -67,10 +68,13 @@ class ActivityStore {
         supportersAvatar.push(constructPicUrl(singlePlayer['icon']))
       }
 
+      if (supportersAvatar.length > 5) {
```
Comment (Keith-CY) [inline src/stores/activityStore/index.js]: if `length <= 5`, `supportersAvatar.slice(0, 5)` works perfectly.

File: src/stores/checkinStore/index.js [hunk 3]
```diff
@@ -49,6 +49,17 @@ class CheckinStore {
       return userContract.methods.getUserClubs(sender, index).call()
     }))
 
+    let clubJoined = []
+    for (let i = 0; i < this.clubAddrList.length; i++) {
+      const dataContract = new appchain.base.Contract(dataAbi, this.clubAddrList[i])
+      const isSignup = await dataContract.methods.signUps(sender).call()
```
Comment (Keith-CY) [inline src/stores/checkinStore/index.js]: It's synchronous, which makes it slow.

**Comment by QingyangKong** [comment]
Use promise.all() to change code from sync to async. @Keith-CY

**Review by Keith-CY — APPROVED** [review]

**Pull request merged**